Docs: mark up json.dump() using parameter list (#128482)
python/cpython: Doc/library/json.rst

      so trying to serialize multiple objects with repeated calls to
      :func:`dump` using the same *fp* will result in an invalid JSON file.

   :param object obj:
      The Python object to be serialized.

   :param fp:
      The file-like object *obj* will be serialized to.
      The :mod:`!json` module always produces :class:`str` objects,
      not :class:`bytes` objects,
      therefore ``fp.write()`` must support :class:`str` input.
   :type fp: :term:`file-like object`

   :param bool skipkeys:
      If ``True``, keys that are not of a basic type
      (:class:`str`, :class:`int`, :class:`float`, :class:`bool`, ``None``)
      will be skipped instead of raising a :exc:`TypeError`.
      Default ``False``.

   :param bool ensure_ascii:
      If ``True`` (the default), the output is guaranteed to
      have all incoming non-ASCII characters escaped.
      If ``False``, these characters will be outputted as-is.

   :param bool check_circular:
      If ``False``, the circular reference check for container types is skipped
      and a circular reference will result in a :exc:`RecursionError` (or worse).
      Default ``True``.

   :param bool allow_nan:

>       If ``False``, serialization of out of range :class:`float` values

Reviewer: ```suggestion
      If ``False``, serialization of out-of-range :class:`float` values
```

      (``nan``, ``inf``, ``-inf``) will result in a :exc:`ValueError`,

>       in strict compliance of the JSON specification.

Reviewer: ```suggestion
      in strict compliance with the JSON specification.
```

      If ``True`` (the default), their JavaScript equivalents
      (``NaN``, ``Infinity``, ``-Infinity``) are used.

   :param indent:
      If a positive integer or string, JSON array elements and
      object members will be pretty-printed with that indent level.

>       A positive integer indent, indents that many spaces per level;

Reviewer: I think we can shorten this a bit:
```suggestion
      A positive integer indents that many spaces per level;
```

>       a string (such as ``"\t"``), uses that string to indent each level.

Reviewer: And maybe?
```suggestion
      a string (such as ``"\t"``) is used to indent each level.
```

      If zero, negative, or ``""`` (the empty string),
      only newlines are inserted.
      If ``None`` (the default), the most compact representation is used.
   :type indent: int | str | None

   :param separators:
      A two-tuple: ``(item_separator, key_separator)``.
      If ``None`` (the default), *separators* defaults to
      ``(', ', ': ')`` if *indent* is ``None``,
      and ``(',', ': ')`` otherwise.

>       For the most compact JSON representation,

Reviewer: Can maybe skip this too?
```suggestion
      For the most compact JSON,
```
